### PR TITLE
versions: Update nydus version to 2.2.1

### LINKED ISSUE #6938: update nydus version to 2.2.1

### PR DESCRIPTION
**Pull request by GabyCT**
This PR updates the nydus version to 2.2.1. This change includes: 
nydus-image: fix a underflow issue in get_compressed_size()
backport fix/feature to stable 2.2
[backport] contrib: upgrade runc to v1.1.5
service: add README for nydus-service
nydus: fix a possible panic caused by SubCmdArgs::is_present Backports two bugfixes from master into stable/v2.2
 [backport stable/v2.2] action: upgrade golangci-lint to v1.51.2 
[backport] action: fix smoke test for branch pattern

Fixes #6938

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @GabyCT!

**Comment by fidencio** [comment]
Please, wait till we cut the release, and only merge this after that.

**Comment by fidencio** [comment]
You can go ahead to merge this one as soon as you'd like. :-)

**Review by cmaf — APPROVED** [review]

**Pull request merged**
